service-wfs-client/src/test/java/org/oskari/service/wfs/client/OskariWFS3ClientTest.java

```diff
@@ -0,0 +1,42 @@
+package org.oskari.service.wfs.client;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+import com.vividsolutions.jts.geom.Envelope;
+
+import fi.nls.oskari.service.ServiceException;
+
+public class OskariWFS3ClientTest {
+
+    @Ignore("Depends on outside service, results might vary")
+    @Test
+    public void testGetFeatures() throws ServiceException, IOException {
+        String endPoint = "https://beta-paikkatieto.maanmittauslaitos.fi/geographic-names/wfs3/v1/";
+        String user = null;
+        String pass = null;
+        String collectionId = "places";
+        CoordinateReferenceSystem crs = OskariWFS3Client.getCRS84();
+        Envelope envelope = new Envelope(21.35, 21.40, 61.35, 61.40);
+        ReferencedEnvelope bbox = new ReferencedEnvelope(envelope, crs);
+        int limit = 10;
```

Comment:
Limit is more like page size since i is expected to be 29?

Reply:
Yea limit is the upper limit for a single request. If there's a link to next page we will follow it. Maybe we another, hard limit for the number of features we want to read (probably we want to stop before reading billions of features). Or maybe we should let the frontend to decide, and paginate all the way to the frontend, that could be interesting. Obviously for MVT tiles that's not what we want, but for GeoJSON maybe.